defaults (#2448)
GMLC-TDC/HELICS: docs/references/configuration_options_reference.md

---

### `default`

_API:_ `helicsInputSetDefaultX`
[C++ input](https://docs.helics.org/en/latest/doxygen/classhelics_1_1Input.html#a55056ac9dd2895270f575827dd9951c7)
| [C input](api-reference/C_API.md#input)
| [Python input](https://python.helics.org/api/capi-py.html#helicsInputSetDefault)
| [Julia input](https://julia.helics.org/latest/api/#HELICS.helicsInputSetDefulat-Tuple{HELICS.Input,XXX})

> This option allows the specific of the default value used when no publication has been received. Each datatype has its own API call such as:

Reviewer: ```suggestion
This option allows specifying the default value used when no publication has been received. Each datatype has its own API call such as:
```

- `helicsInputSetDefaultBoolean()`
- `helicsInputSetDefaultBytes()`
- `helicsInputSetDefaultChar()`
- `helicsInputSetDefaultComplex()`
- `helicsInputSetDefaultComplexVector()`
- `helicsInputSetDefaulDouble()`
- `helicsInputSetDefaultInteger()`
- `helicsInputSetDefaultRaw()`
- `helicsInputSetDefaultString()` (used for JSONs)
- `helicsInputSetDefaultVector()`

> Note that the following, though they have syntaxical similarities to the above APIs, **do not** set default values on HELICS inputs when no publications have been received.

Reviewer: ```suggestion
Note that the following, though they have syntactical similarities to the above APIs, **do not** set default values on HELICS inputs when no publications have been received.
```

> - `helicsInputSetDefaultTime()` - int64 but can also take a double

Author: ```suggestion
- `helicsInputSetDefaultTime()` - set the default using a HelicsTime value
```

> - `helicsInputSetDefaultNamedPoint()`

Author: Are these not working,  this should do exactly the same thing as the others?  What makes you think they don't?

Reviewer 2: As far as I know they are working fine they just aren't used to set the default value for an input that hasn't been published to.

> - `helicsInputSetDefaultNamedPoint()` - string or a double

Author: ```suggestion
- `helicsInputSetDefaultNamedPoint()` -set the default NamedPoint which is a pair of a string and double used for tagged values or set points
```
